azure-proto-network/Extensions/ResourceGroupExtension.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Azure;
+using Azure.ResourceManager.Network.Models;
+using azure_proto_core;
+using azure_proto_core.Resources;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace azure_proto_network
+{
+    public static class ResourceGroupExtension
```

Comment:
Note: I think the class name here should be plural (which the previous one also was not)